src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/Options/OptionsExtensions.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Microsoft.Diagnostics.Monitoring.UnitTests.Options
+{
+    internal static class OptionsExtensions
+    {
+        /// <summary>
+        /// Generates an environment variable map of the options.
+        /// </summary>
+        /// <remarks>
+        /// Each key is the variable name; each value is the variable value.
+        /// </remarks>
+        public static IDictionary<string, string> ToEnvironmentConfiguration(this RootOptions options)
+        {
+            Dictionary<string, string> variables = new Dictionary<string, string>();
+            MapObject(options, "DotNetMonitor_", variables);
+            return variables;
+        }
+
+        /// <summary>
+        /// Generates a key-per-file map of the options.
+        /// </summary>
+        /// <remarks>
+        /// Each key is the file name; each value is the file content.
+        /// </remarks>
+        public static IDictionary<string, string> ToKeyPerFileConfiguration(this RootOptions options)
+        {
+            Dictionary<string, string> variables = new Dictionary<string, string>();
+            MapObject(options, string.Empty, variables);
```

Comment:
Will this work properly for array based options? E.g. Something__0__Name, Something__1__Name

Reply:
I believe that `MapList` is doing this. We can add other type handling in the future if we use other list types in the configuration options.